src/prefect/utilities/storage.py

```diff
@@ -143,3 +151,92 @@ def extract_flow_from_module(module_str: str, flow_name: str = None) -> "Flow":
                 return attr
 
     raise ValueError("No flow found in module.")
+
+
+def flow_to_bytes_pickle(flow: "Flow") -> bytes:
+    """Serialize a flow to bytes.
+
+    The flow is serialized using `cloudpickle`, with some extra metadata on
+    included via JSON. The flow can be reloaded using `flow_from_bytes_pickle`.
+
+    Args:
+        - flow (Flow): the flow to be serialized.
+
+    Returns:
+        - bytes: a serialized representation of the flow.
+    """
+    flow_data = binascii.b2a_base64(
+        cloudpickle.dumps(flow, protocol=4), newline=False
+    ).decode("utf-8")
+    out = json.dumps({"flow": flow_data, "versions": _get_versions()})
+    return out.encode("utf-8")
+
+
+def _get_versions() -> dict:
+    """Get version info on libraries where a version-mismatch between
+    registration and execution environment may cause a flow to fail to load
+    properly"""
+    return {
+        "cloudpickle": cloudpickle.__version__,
+        "prefect": prefect.__version__,
+        "python": "%d.%d.%d" % sys.version_info[:3],
+    }
+
+
+def flow_from_bytes_pickle(data: bytes) -> "Flow":
+    """Load a flow from bytes."""
+    try:
+        info = json.loads(data.decode("utf-8"))
+    except Exception:
+        # Serialized using older version of prefect, use cloudpickle directly
+        flow_bytes = data
+        reg_versions = {}
+    else:
+        flow_bytes = binascii.a2b_base64(info["flow"])
+        reg_versions = info["versions"]
+
+    run_versions = _get_versions()
+
+    try:
+        flow = cloudpickle.loads(flow_bytes)
+    except Exception as exc:
+        parts = ["An error occurred while unpickling the flow:", f"  {exc!r}"]
+        # Check for mismatched versions to provide a better warning if possible
+        mismatches = []
+        for name, v1 in sorted(reg_versions.items()):
+            if name in run_versions:
+                v2 = run_versions[name]
+                if LooseVersion(v1) != v2:
+                    mismatches.append(
+                        f"  - {name}: (flow built with {v1!r}, currently running with {v2!r})"
+                    )
+        if mismatches:
+            parts.append(
+                "This may be due to one of the following version mismatches between "
+                "the flow build and execution environments:"
+            )
+            parts.extend(mismatches)
+        if isinstance(exc, ImportError):
```

Comment:
Is there a way to relay what package is actually causing the error?

Reply:
Not in a way I'd like to do. The original error is still displayed, but inferring the package from the message is error prone. Further, the user may be missing several dependencies - elevating one (the first import error) feels incorrect to me.